### PR TITLE
{chem}[intel/2017b] ABINIT v8.4.4

### PR DESCRIPTION
**Pull request by migueldiascosta**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by migueldiascosta** [comment]
requires #5249 and #5250 (ABINIT now supports libxc 3.0, https://www.abinit.org/package/8.4.4)

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
grc-cluster1 - Linux centos 6.9, Intel(R) Xeon(R) CPU E5-2640 0 @ 2.50GHz, Python 2.6.6
See https://gist.github.com/c209b857a9983786a07f79b25c4bc9a2 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2520.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/86a6b244ac4eef7070875bd58ac091e6 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2443.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/93d0eaaf9bee1fa563027afa380a7c4f for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 2 out of 3 (1 easyconfigs in this PR)
node2047.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/74ae7f9f0a0d6657937ff18bc00070a2 for a full test report.

**Comment by migueldiascosta** [comment]
@boegel I can't reproduce your failing tests, any idea what's going on there?

**Comment by boegel** [comment]
@migueldiascosta I have to say I'm pretty clueless here...

The error is quite clear: `forrtl: severe (71): integer divide by zero`.

This is the last bit of output for one of the failing tests (`fast_t30`):

```
 ==== FFT mesh ====
  FFT mesh divisions ........................    10   10   10
  Augmented FFT divisions ...................    11   11   10
  FFT algorithm .............................   112
  FFT cache size ............................    16
```

Not sure if that's helpful at all...

**Comment by boegel** [comment]
One of our colleagues just reported the exact same issue (division by zero with ABINIT), and has pinpointed it to https://software.intel.com/en-us/articles/inconsistent-program-behavior-on-red-hat-enterprise-linux-74-if-compiled-with-intel .

A test install with `export LD_BIND_NOW=1` is now running, report will be submitted in a bit.

For the Intel article, it seems like you may be able to dance around this by building with `-fPIC` enabled, since the Intel compilers included in `intel/2017b` are version 17.0 update 4 (`2017.4.196`)...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2637.swalot.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/422f96c8738328231cfe0a5cadc7cb42 for a full test report.

File: easybuild/easyconfigs/a/ABINIT/ABINIT-8.4.4-intel-2017b.eb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+# netcdf support
+configopts += '--with-netcdf-incs="-I$EBROOTNETCDF/include -I$EBROOTNETCDFMINFORTRAN/include" '
+configopts += '--with-netcdf-libs="-L$EBROOTNETCDF/lib -lnetcdf -L$EBROOTNETCDFMINFORTRAN/lib -lnetcdff" '
+
+dependencies = [
+    ('libxc', '3.0.0'),
+    ('netCDF', '4.4.1.1'),
+    ('netCDF-Fortran', '4.4.4'),
+]
```
Comment (boegel) [inline easybuild/easyconfigs/a/ABINIT/ABINIT-8.4.4-intel-2017b.eb]: @migueldiascosta Do you mind moving this up, above `configopts =`?

File: easybuild/easyconfigs/a/ABINIT/ABINIT-8.4.4-intel-2017b.eb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+version = "8.4.4"
+
+homepage = 'http://www.abinit.org/'
+description = """ABINIT is a package whose main program allows one to find the total energy,
+ charge density and  electronic structure of systems made of electrons and nuclei (molecules
+ and periodic solids) within Density Functional  Theory (DFT), using pseudopotentials and a
+ planewave or wavelet basis."""
+
+toolchain = {'name': 'intel', 'version': '2017b'}
+toolchainopts = {'usempi': True}
```
Comment (boegel) [inline easybuild/easyconfigs/a/ABINIT/ABINIT-8.4.4-intel-2017b.eb]: Please also enable the use of `-fPIC` (to dance around the problem with the Intel compilers & recent `glibc` versions).

```python
toolchainoptions = {'usempi': True, 'pic': True}
```

I double-checked, this also fixes the problem with the tests, just like defining `$LD_BIND_NOW` does...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2455.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/bfb24ec23f04a7d1fb4ca03807253b18 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
gligar03.gligar.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/03f9046157192217cc3c10df4adb9113 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @migueldiascosta!

**Comment by vanzod** [comment]
@migueldiascosta I have been comparing this easyconfig with the ones for previous versions of `ABINIT` and I noticed that `--with-trio-flavor` has been removed from `configopts` while `--with-linalg-flavor` and `--with-fft-flavor` have been added. Can you please explain the changes? Thanks

**Comment by migueldiascosta** [comment]
@vanzod without the explicit `linalg` and `fft` flavors, it can link against system libraries if present 

regarding `trio`, and in particular `etsf_io` support, I missed that, should we fix it in the existing easyconfigs?

**Comment by vanzod** [comment]
@migueldiascosta Thanks for the clarification. For the missing `etsf_io` support, fix it in the easyconfig and open a new PR.

**Comment by vanzod** [comment]
@migueldiascosta I meant a bugfix PR for all the easyconfigs that are missing it. Sorry for the confusion.

**Comment by migueldiascosta** [comment]
@vanzod the `etsf_io` flag doesn't work anymore, it seems it is enough to specify `--with-trio-flavor=netcdf`

```
The ETSF_IO library is not used anymore : 
it has been replaced by an internal NetCDF implementation 
of the ETSF File Format Specification in Abinit. 
The support of the ETSF_IO library will soon end.

IMPORTANT : Do not be mistaken ! 
The ETSF File Format Specification is used for most
ABINIT NetCDF files and will continue to be so
in the foreseen future.

Please contact Matteo Giantomassi for any question about the
new implementation of the reading and writing of the 
ETSF formatted files.
```

also, my 7.10.5 easyconfigs didn't even have `netcdf` as a dependency...

I'll create a bugfix PR to add `--with-trio-flavor=netcdf` to the 8.4.4 and 8.6.3 easyconfigs

**Comment by akesandgren** [comment]
Yes, etsf_io disappeared in v8.4 at least.

**Pull request merged**